src/main.js

```diff
@@ -174,6 +183,24 @@ async function onReady() {
 	try {
 		const [cachedDataDir, nlsConfig] = await Promise.all([nodeCachedDataDir.ensureExists(), resolveNlsConfiguration()]);
 
+		protocol.registerFileProtocol('vscode-file', async (request, callback) => {
```

Comment:
@deepak1556 I feel like this (`registerFileProtocol` and `interceptFileProtocol`) could move into `app.ts`. I am sensitive to adding code to `main.js` because it is being hit even if you start a second instance from the terminal that talks to the first. It seems to me this protocol handler is only needed when opening a window.

One place where it could fit is before we open the window:

https://github.com/microsoft/vscode/blob/003a3678c879ee5e9a72ad9e5088d89672d75bd8/src/vs/code/electron-main/app.ts#L397